test/unit/rules/attributes-order-test.js

```diff
@@ -0,0 +1,310 @@
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        alphabetize: false,
+      },
+      template:
+        '<MyComponent @value="5" data-test-foo @change={{this.foo}} local-class="foo" {{on "click" this.foo}} ...attributes as |sth|>content</MyComponent>',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 38,
+              "endColumn": 145,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "line": 1,
+              "message": "Argument @change={{this.foo}} must go before attributes, modifiers and splattributes",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "@change={{this.foo}}",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        attributeOrder: ['attributes', 'arguments', 'modifiers', 'splattributes'],
+      },
+      template: '<div @foo="1" {{did-render this.ok}} ...attributes aria-label="foo"></div>',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 51,
+              "endColumn": 74,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "line": 1,
+              "message": "Attribute aria-label=\\"foo\\" must go before arguments, modifiers and splattributes",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "aria-label=\\"foo\\"",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        alphabetize: true,
+      },
+      template:
+        '<div @foo="1" {{did-update this.notok}} {{did-render this.ok}} ...attributes aria-label="foo"></div>',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 40,
+              "endColumn": 100,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "line": 1,
+              "message": "Modifier {{did-render this.ok}} is not alphabetized",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "{{did-render this.ok}}",
+            },
+          ]
+        `);
+      },
+    },
+    {
+      config: {
+        alphabetize: true,
+      },
+      template: '{{MyComponent b="2" a="1"}}',
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 20,
+              "endColumn": 27,
+              "endLine": 1,
+              "filePath": "layout.hbs",
+              "line": 1,
+              "message": "Attribute a=\\"1\\" is not alphabetized",
+              "rule": "attributes-order",
+              "severity": 2,
+              "source": "a=\\"1\\"",
+            },
+          ]
+        `);
+      },
+    },
+  ],
+
+  error: [
```

Comment:
We should also have error test cases with invalid values for the actual options:

* `attributeOrder: ['arguments', 'NOT_AN_OPTION']`
* `alphabetize: 123`